### ISSUE TITLE
Restrict "flat" format to architectures that support it (was: elf2flt.c:126:2: error: #error "Don't know how to support your CPU architecture??")

### ISSUE THREAD

**Issue by 8ohmh** [crosstool-ng/crosstool-ng#878]
Hi again!

Now trying compile for an AVR Board but get:

`elf2flt.c:126:2: error: #error "Don't know how to support your CPU architecture??"`

Crosstools-NG info:
```
crosstool-ng-1.23.0-250-g8c581c25
i86_64, debian 9.2 stretch Kernel 4.9.2 or so...
```

Config and build log:
[avr-failed.tar.gz](https://github.com/crosstool-ng/crosstool-ng/files/1531958/avr-failed.tar.gz)
87e82b9aff110037982f1877d3182614  ../avr-failed.tar.gz

**Comment by stilor** [comment]
Architectures without MMU need special "flat" binaries; elf2flt is needed to support it. If you feel it should, submit a but to that project: https://github.com/uclinux-dev/elf2flt; but last I checked, there was no Linux for 8-bit AVR.

I am also not sure how you ended up with this configuration: crosstool-ng does not let one select uclibc-ng as the C library for AVR; neither does it allow you to select Linux kernel as the OS. Did you manually edit the .config file?

**Comment by 8ohmh** [comment]
Seems like that it is something comnparable to the ARM Problem: `"not passing -mcpu param to gcc sub processes..." `I tried to do it somehow that I can compile "somehow successfully" till end... (after the same fifty or so config guessings like in my ARM compiling tries...)   

BTW 32 Bit AVR not available?

**Comment by stilor** [comment]
I am not sure what "ARM Problem" you are talking about. A reference/link would help.

AVR32 has been retired by @bhundven awhile ago. Perhaps he can jump in to explain the reasons.

You haven't answered my question how you managed to produce this .config that selects linux/uclibc-ng for AVR.

**Comment by 8ohmh** [comment]
1. https://github.com/crosstool-ng/crosstool-ng/issues/857
2. Changing e.g. the .config file, then the kernel.in file etc...

 am trying every possible (and even not possible config) till I get a completely working crossrtool-ng process till end ... (I know there are not possible configs, but as I tried in point 1 that even working examples are NOT working correctly on my PC I am doing this way too)

**Comment by 8ohmh** [comment]
Hi, why is my last avr error issue deleted ?

**Comment by stilor** [comment]
Really, "changing kernel.in"? if there are some dependencies in it, they are there for a reason. You change the sources, fine, this is what open source is for - but please do your own debugging in that case.

There is a valid problem here, crosstool-NG should not allow you to select "flat" format for architectures that do not support it; AVR included.

Your other issue was not *deleted* but rather *closed*. Please take a moment to read why.

**Comment by 8ohmh** [comment]
Thanks for the answer, and - of course in first line my changings don't belong to main git branch of crosstools (I know in that case I have to do a branch clone). Beside I meant the other (deleted) issue (not that here) which was using almost the default config for avr-8 - why was this deleted?

But till now I have never ever got a working correctly toolchain compilation on the NAS/ARM board (dmesg is put in point 1 of https://github.com/crosstool-ng/crosstool-ng/issues/857 - we discussed about to put a bug report for the GMP/GCC guys, but this take time, till the changes are in production)

And I was expecting that corsstolos-ng are working by default vaules even on such boards..,

**Comment by stilor** [comment]
We don't delete issues. What was the issue ID (you should still have that in the emails). I see 4 issues by you, 3 closed (#857, #878, #879) and one open (#867).

BTW, I am going to keep this open - I still need to do what I said above ("crosstool-NG should not allow you to select "flat" format for architectures that do not support it").

**Comment by bhundven** [comment]
https://sourceware.org/ml/crossgcc/2015-10/msg00050.html
